src/LinAlg/hiopMatrixRajaSparseTriplet.hpp

```diff
@@ -143,75 +173,82 @@ class hiopMatrixRajaSparseTriplet : public hiopMatrixSparse
    * the (strictly) lower triangular  elements (these are ignored later on since only the upper 
    * triangular part of W will be accessed)
    */
-  virtual void addMDinvNtransToSymDeMatUTri(int row_dest_start, int col_dest_start, const double& alpha,
-					    const hiopVector& D, const hiopMatrixSparse& N,
-					    hiopMatrixDense& W) const;
+  virtual void addMDinvNtransToSymDeMatUTri(int row_dest_start,
+                                            int col_dest_start,
+                                            const double& alpha,
+                                            const hiopVector& D,
+                                            const hiopMatrixSparse& N,
+                                            hiopMatrixDense& W) const;
 
   virtual void copyRowsBlockFrom(const hiopMatrix& src_gen,
                                  const index_type& rows_src_idx_st,
                                  const size_type& n_rows,
                                  const index_type& rows_dest_idx_st,
-                                 const size_type& dest_nnz_st)
-  {
-    assert(false && "not needed / implemented");
-  }
-
+                                 const size_type& dest_nnz_st);
+  
+  /**
+  * @brief Copy matrix 'src_gen', into 'this' as a submatrix from corner 'dest_row_st' and 'dest_col_st'
+  * The non-zero elements start from 'dest_nnz_st' will be replaced by the new elements. 
+  * When `offdiag_only` is set to true, only the off-diagonal part of `src_gen` is copied.
+  *
+  * @pre 'this' must have enough rows and cols after row 'dest_row_st' and col 'dest_col_st'
+  * @pre 'dest_nnz_st' + the number of non-zeros in the copied matrix must be less or equal to 
+  * this->numOfNumbers()
+  * @pre User must know the nonzero pattern of src and dest matrices. The method assumes 
+  * that non-zero patterns does not change between calls and that 'src_gen' is a valid
+  *  submatrix of 'this'
+  */
   virtual void copySubmatrixFrom(const hiopMatrix& src_gen,
                                  const index_type& dest_row_st,
                                  const index_type& dest_col_st,
                                  const size_type& dest_nnz_st,
-                                 const bool offdiag_only = false)
-  {
-    assert(false && "not needed / implemented");
-  }
+                                 const bool offdiag_only = false) {assert(0 && "FIXME_NY");}
```

Comment:
fix assert message and code styling.

Reply:
These functions require RAJA implementations, which are filled in other PRs.
I changed the description in assert, to highlight these functions and hence I can quickly search for them.
For example, this function `copySubmatrixFrom` is implemented in `sparse-gpu-kernel-2-dev`, and you can [see this assert is removed from the code](https://github.com/LLNL/hiop/blob/6cf5c1087378c9b478cb38ced32f569ace399791/src/LinAlg/hiopMatrixRajaSparseTriplet.hpp#L202-L206).

Reply:
ok, I see. But the ones below are not removed in `sparse-gpu-kernel-2-dev. My point was to use a more suggestive assert message "not yet implemented".

Reply:
They are removed in sparse-gpu-kernel-3-dev and sparse-gpu-kernel-4-dev. :-)
I will change the message to "not yet implemented" and rebase the following branches.